api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -556,8 +582,11 @@ type HostedClusterSpec struct {
 	// +immutable
 	OLMCatalogPlacement OLMCatalogPlacement `json:"olmCatalogPlacement,omitempty"`
 
-	// NodeSelector when specified, must be true for the pods managed by the HostedCluster to be scheduled.
-	//
+	// NodeSelector when specified, is propagated to all hostedCluster Deployments and Stateful sets running management side.
+	// It must be satisfied by the management Nodes for the pods to be scheduled. Otherwise the HostedCluster will enter a degraded state.
+	// Changes to this field will propagate to existing Deployments and StatefulSets.
+	// +kubebuilder:validation:XValidation:rule="size(self) <= 50",message="nodeSelector map can have at most 5 entries"
+	// TODO(alberto): add additional validation for the map key/values.
```

Comment:
nit: would be nice to link a Jira ticket here so we don't lose track of this TODO.

Reply:
This will be a different activity